src/solver/hydro/management/management.cpp

```diff
@@ -290,43 +290,31 @@ bool HydroManagement::checkWeeklyMinGeneration(uint tsIndex, Data::Area& area) c
     return true;
 }
 
-bool HydroManagement::checkHourlyMinGeneration(uint tsIndex, Data::Area& area) const
+bool HydroManagement::checkHourlyMinMaxGeneration(uint tsIndex, uint tsIndexPowerCredits, Data::Area& area) const
 {
     // Hourly minimum generation <= hourly inflows for each hour
-    const auto& calendar = study.calendar;
     auto& mingenmatrix = area.hydro.series->mingen;
     auto const& srcmingen = mingenmatrix[tsIndex < mingenmatrix.width ? tsIndex : 0];
-    auto const& maxPower = area.hydro.maxPower;
-    auto const& maxP = maxPower[Data::PartHydro::genMaxP];
+    auto& maxgenmatrix = area.hydro.series->maxgen;
+    auto const& srcmaxgen
+      = maxgenmatrix[tsIndexPowerCredits < maxgenmatrix.width ? tsIndexPowerCredits : 0];
 
-    if (!area.hydro.reservoirManagement)
```

Comment:
This condition : `if (!area.hydro.reservoirManagement)` disappeared. It is expected ?

Reply:
This can be resolved.